pkg/generate/app/cmd/newapp.go

```diff
@@ -1113,6 +1112,25 @@ func (c *AppConfig) run(acceptors app.Acceptors) (*AppResult, error) {
 	}, nil
 }
 
+// checkCircularReferences ensures there are no builds that can trigger themselves
+// due to an imagechangetrigger that matches the output destination of the image.
```

Comment:
I think just because something is not exported is not an excuse to not document or make things clear.
And if something is missing elsewhere, it doesn't mean we cannot do better.

AFAIK, we're not generating "readable" godocs like one would read for the standard library or other libs we use, but I believe all devs are consuming those docs straight from code every day.

The type is explicit, we don't need to repeat ourselves, but intentions are not always obvious.

For instance, when reading this piece I first wondered why it operates on `app.Objects` and not on a single `api.BuildConfig`. It takes extra leaps of thought and clicking, digging, etc, to figure out what is `app.Objects` and what it actually might contain.

I thought having this type of discussion was beneficial to a code review. After all, we're all going to maintain this code later.